### ISSUE TITLE
Misleading usage of arguments when estimating from checkpoint

### ISSUE THREAD

**Issue by justinian336** [sansan-inc/lighthergm#38]
### Problem

When starting an estimation of `lighthergm::hergm` from a previous checkpoint, the model uses the settings from the checkpoint, even if you explicitly set different arguments.

This is intentional:
https://github.com/sansan-inc/lighthergm/blob/965686bb2df3ddf1fa3cae135866dedf9495319b/lighthergm/R/hergm.R#L161

However, that's a misleading API. Normally, it is expected that arguments that are set explicitly take precedence over the implicitly set ones.

### Expected behavior
If `checkpoint_1` in the following code was obtained with 3,000 iterations, and you attempt to estimate `checkpoint_2` the following way:
```R
checkpoint_2 <- lighthergm::hergm(
      checkpoint_1,
      n_em_step_max = 92
    )
```
Then the estimation should use 92 iterations instead of 3,000. Same logic for other arguments that can be set explicitly. If for some reason that behavior is not desirable, display a message for the user to know that the argument they set explicitly won't be used.

**Comment by meleangelo** [comment]
I usually just change the variable in the object. For example, if the object is checkpoint_1, then you can change checkpoint_1$n_em_step_max to whatever you want. I believe this is a simple fix, but in the meantime, it should work.

**Comment by justinian336** [comment]
Yeah, it is possible to do it like that. I think that we could come up with a more intuitive interface for a future update.
